cpp/src/DataStorm/NodeSessionI.cpp

```diff
@@ -26,17 +27,17 @@ namespace
     public:
         NodeForwarder(
             shared_ptr<NodeSessionManager> nodeSessionManager,
-            shared_ptr<NodeSessionI> nodeSession,
+            const shared_ptr<NodeSessionI>& nodeSession,
             NodePrx node)
             : _nodeSessionManager(std::move(nodeSessionManager)),
-              _nodeSession(std::move(nodeSession)),
+              _nodeSession(nodeSession),
```

Comment:
`_nodeSession` is a weak_ptr. The weak_ptr constructor takes a const reference to a shared_ptr.